routes/transaction.go

```diff
@@ -122,6 +122,116 @@ func (fes *APIServer) GetTxn(ww http.ResponseWriter, req *http.Request) {
+		return
+	}
+	if len(requestData.TxnHashHexes) > GetTxnsRequestCountMax {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetTxns: TxnHashHexes length %d is greater than %d.",
+			len(requestData.TxnHashHexes),
+			GetTxnsRequestCountMax,
+		))
+		return
+	}
+
+	// Validate the TxnStatus param.
+	txnStatus := requestData.TxnStatus
+	if txnStatus == "" {
+		txnStatus = TxnStatusInMempool
+	}
+	if txnStatus != TxnStatusInMempool && txnStatus != TxnStatusCommitted {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetTxns: Invalid TxnStatus: %s. Options are {InMempool, Committed}.", txnStatus,
+		))
+		return
+	}
+
+	// Decode the TxnHashHexes.
+	var txnHashes []*lib.BlockHash
+	for _, txnHashHex := range requestData.TxnHashHexes {
+		txnHashBytes, err := hex.DecodeString(txnHashHex)
+		if err != nil || len(txnHashBytes) != lib.HashSizeBytes {
+			_AddBadRequestError(ww, fmt.Sprintf("GetTxns: Error parsing txn hash %s: %v", txnHashHex, err))
+			return
+		}
+		txnHashes = append(txnHashes, lib.NewBlockHash(txnHashBytes))
+	}
+
+	// The order of operations is tricky here. We need to do the following in this
+	// exact order:
+	// 1. Check the mempool for each txn
+	// 2. Wait for txindex to fully sync
+	// 3. Then check txindex
+	//
+	// If we instead check the mempool afterward, then there is a chance that a txn
+	// has been removed by a new block that is not yet in txindex. This would cause the
+	// endpoint to incorrectly report that the txn doesn't exist on the node, when in
+	// fact it is in "limbo" between the mempool and txindex.
+	//
+	// 1. Check the mempool for each txn if TxnStatusInMempool.
+	txnsFoundInMempool := make(map[string]bool)
+	if txnStatus == TxnStatusInMempool {
+		mempool := fes.backendServer.GetMempool()
+		for _, txnHash := range txnHashes {
+			txnsFoundInMempool[txnHash.String()] = mempool.IsTransactionInPool(txnHash)
+		}
+	}
+
+	// 2. We have to wait until txindex has reached the uncommitted tip height, not
+	//    the committed tip height. Otherwise, we'll be missing ~2 blocks in limbo.
+	startTime := time.Now()
+	coreChainTipHeight := fes.TXIndex.CoreChain.BlockTip().Height
+	for fes.TXIndex.TXIndexChain.BlockTip().Height < coreChainTipHeight {
+		if time.Since(startTime) > 30*time.Second {
+			_AddBadRequestError(ww, fmt.Sprintf("GetTxns: Timed out waiting for txindex to sync."))
+			return
+		}
+		time.Sleep(10 * time.Millisecond)
+	}
+
+	// 3. Check the txindex for each txn.
+	txnsFoundInTxindex := make(map[string]bool)
+	for _, txnHash := range txnHashes {
+		txnsFoundInTxindex[txnHash.String()] = lib.DbCheckTxnExistence(fes.TXIndex.TXIndexChain.DB(), nil, txnHash)
+	}
+
+	// Construct response.
+	res := &GetTxnsResponse{TxnsFound: make(map[string]bool)}
+	for _, txnHash := range txnHashes {
+		// If TxnStatusCommited, txnsFoundInMempool[txnHash.String()] will always be false.
+		res.TxnsFound[txnHash.String()] = txnsFoundInMempool[txnHash.String()] || txnsFoundInTxindex[txnHash.String()]
+	}
```

Comment:
if `TxnStatus` is `Committed`, then we can simply return `txnsFoundInTxindex`. We only need to do this merge if `TxnStatus === InMempool`. And if we need to do the merge, then we can use `TxnHashHex` instead of `.String()`.